File: Java/Stack.java

public class Stack {
     private static class Node {
        int value;
        Node next;
        public Node(int val) {
            this.value = val;
        }
    }
    private int height;
    private Node top;
    public void push(int val) {
        Node new_node = new Node(val);
        if(height == 0) {
            top = new_node;
        } else {
            new_node.next = top;
            top = new_node;
        }
        height++;
    }
    public int pop() {
        if(height == 0) {
            return -1;
        } else if(height == 1) {
            int val = top.value;
            top = null;
            height--;
            return val;
        } else {
            int val = top.value;
            Node temp = top.next;
            top.next = null;
            top = temp;
            height--;
            return val;
        }
    }
    public int size() {
        return height;
    }
    @Override
    public String toString() {
        Node temp = top;
        String str = "";
        while (temp != null) {
            str += temp.value + ", ";
            temp = temp.next;
        }
        if(!str.equals("")) {
            str = "[" + str.substring(0,str.length() - 2) + "]";
        } else {
            str = "[]";
        }
        return  str;
    }
}

File: C++/stack.cpp
#include "stack.h"
#include <iostream>
using std::vector;
using std::cerr;
stack::stack()
{
    length = 0;
    top = nullptr;
}

stack::~stack()
{
    if(length == 0) cerr << "THE STACK IS ALREADY EMPTY";
    node* cur = top;
    while(cur != nullptr) {
        node* temp= cur->below;
        delete cur;
        cur = temp;
    }
}
void stack::push(int val) {
    node* nd = new node(val);
    if(length == 0) top = nd;
    nd->below = top;
    top = nd;
    length++;
    
}
int stack::pop() {
    if(length == 0) cerr << "THE STACK IS EMPTY";
    int rt_val = top->val;
    node* temp = top;
    top = top->below;
    delete temp;
    length--;
    return rt_val;
}
vector<int> stack::display() {
    if(length == 0) cerr << "THE STACK IS EMPTY";
    vector<int> arr;
    node* temp = top;
    while(temp != nullptr) {
        arr.push_back(temp->val);
        temp = temp->below;
    }
    return arr;
}